### PR TITLE
Add support for spm

### LINKED ISSUE #199: Add support for Swift Package Manager to manage dependencies
- [x] Add a section about swift package manager (spm) in brief similar to the pods to README.md
- [x] Add Package.swift
   - [x] modify other files to make SPM possible i.e. Application.swift
- [x] Follow the best practice mentioned in [Package Manager](https://swift.org/package-manager/)

### PR DESCRIPTION
**Pull request by akarsh**
This adds SPM support for the Cauli Framework.

Fixes #199

### CODE REVIEW

File: Rakefile
Comment (brototyp) [inline Rakefile:44]: The current build on CI here is due to a copy-past issue 😊

In the Matomo repository there is a `package_manager` namespace where both the SPM and the Carthage build is tested. Here, the `package_manager` namespace isn't defined and thus `package_manager:spm` is not defined and can't be executed. I think there are two options:
1. Adding a `package_manager` namespace or
2. Changing this line to `Rake::Task['test:spm'].invoke`

I am fine with either. And then we can have a look if just `swift build` is sufficient or if we need to define the sdk and target.

Comment (akarsh) [reply Rakefile:44]: Thanks
I did not work with Rakefile before and did not know much about the ruby syntax here :).

I will go with the first option
>Adding a package_manager namespace or
I find it a good solution to have a separation of the task from the main test.

Comment (akarsh) [reply Rakefile:44]: I did the refactor in this commit 1c37192

**Comment by akarsh** [comment]
>swift build is sufficient or if we need to define the sdk and target.

it is not sufficient anymore. I saw the error from CI - https://travis-ci.org/cauliframework/cauli/builds/595614756#L1985 which is the same as I mentioned before https://github.com/cauliframework/cauli/issues/199#issuecomment-539604961 we need to add some configuration to define the target and sdk path

File: Rakefile
Comment (brototyp) [reply Rakefile:44]: Nice!

**Comment by brototyp** [comment]
> > swift build is sufficient or if we need to define the sdk and target.
> 
> it is not sufficient anymore. I saw the error from CI - https://travis-ci.org/cauliframework/cauli/builds/595614756#L1985 which is the same as I mentioned before [#199 (comment)](https://github.com/cauliframework/cauli/issues/199#issuecomment-539604961) we need to add some configuration to define the target and sdk path

Yeah. I also had a look at it. I thought we can just set the Platform in the `Package.swift` like this (which we should) but that doesn't work either.

```swift
    name: "Cauliframework",
    platforms: [
        .iOS(.v8)
    ],
```

Do you want to try the solution you found before?

**Comment by akarsh** [comment]
In this commit b6f93ba I tried with my solution to the `swift build` it works now.

**Comment by brototyp** [comment]
Super awesome! Great work. What do you think about either tweaking the `Package.swift` to comply with the switflint rules or exclude it from being checked by swiftlint?
Just so we get rid of the warnings here. I think once we got that, we can merge this.

**Comment by akarsh** [comment]
> Super awesome! Great work. What do you think about either tweaking the `Package.swift` to comply with the switflint rules or exclude it from being checked by swiftlint?
> Just so we get rid of the warnings here. I think once we got that, we can merge this.

I added Package.swift in exclude section of swiftlint in this commit 
5269f89

**Review by brototyp — APPROVED** [review]

**Comment by brototyp** [comment]
Really nice. I noticed one more thing. In your ticket you wrote this:

> Add a section about swift package manager (spm) in brief similar to the pods to README.md

I think that's actually a really good idea. Do you still want to do that?

**Comment by akarsh** [comment]
I agree with you. I was waiting for the approval to the code before adding the section to the README.md file. This commit cf7aad5 updates the README.md file.

**Comment by brototyp** [comment]
Really nice. Well done! Thanks for your work!

**Comment by brototyp** [comment]
Hey @akarsh, congratulations to your first merged pull request. Cauli is now your project too. I just invited you as a team member with write permission. We are exited to have you!

**Comment by akarsh** [comment]
Thank you @brototyp 
I am happy to be part of the project.

**Pull request merged**
